contracts/ERC20Safe.sol

```diff
@@ -181,10 +193,14 @@ contract ERC20Safe is BridgeRole, Pausable {
     function initSupply(address tokenAddress, uint256 amount) external onlyAdmin {
         require(whitelistedTokens[tokenAddress], "Unsupported token");
 
-        tokenBalances[tokenAddress] += amount;
-
-        IERC20 erc20 = IERC20(tokenAddress);
-        erc20.safeTransferFrom(msg.sender, address(this), amount);
+        if (!whitelistedTokensMintBurn[tokenAddress]) {
```

Comment:
maybe have some functions called _shouldMint and _shouldBurn? I don't think this condition is readable enough

Reply:
_isTokenMintBurn added